src/components/ClusterOverview/Details/ClusterDetails.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+
+import { ClusterOverviewContextGenericConsumer } from '../ClusterOverviewContext';
+import { InlineLoading } from '../../Loading';
+import { Details } from './Details';
+
+export const ClusterDetails = ({ infrastructure, openshiftVersionResponse, LoadingComponent }) => {
+  const items = {
+    name: {
+      title: 'Name',
+      value: getClusterName(infrastructure),
+      isLoading: !infrastructure,
+    },
+    provider: {
+      title: 'Provider',
+      value: get(infrastructure, 'status.platform'),
+      isLoading: !infrastructure,
+    },
+    rhhi: {
+      title: 'RHHI version',
+      value: 'hardcoded-version', // this will be hardcoded for the demo
+      isLoading: false,
+    },
+    openshift: {
+      title: 'Openshift version',
+      value: getOpenshiftVersion(openshiftVersionResponse),
+      isLoading: !openshiftVersionResponse,
+    },
+  };
+
+  return <Details items={items} LoadingComponent={LoadingComponent} />;
+};
+
+ClusterDetails.defaultProps = {
+  infrastructure: null,
+  openshiftVersionResponse: null,
+  LoadingComponent: InlineLoading,
+};
+
+ClusterDetails.propTypes = {
+  infrastructure: PropTypes.object,
+  openshiftVersionResponse: PropTypes.object,
+  LoadingComponent: PropTypes.oneOfType([PropTypes.node, PropTypes.func]),
+};
+
+export const ClusterDetailsConnected = () => (
+  <ClusterOverviewContextGenericConsumer Component={ClusterDetails} dataPath="detailsData" />
+);
+
+const getClusterName = infrastructure => {
+  const apiServerURL = get(infrastructure, 'status.apiServerURL');
+  let clusterName;
+  if (apiServerURL) {
+    clusterName = apiServerURL.replace('https://api.', '');
+    const portIndex = clusterName.indexOf(':');
+    if (portIndex !== -1) {
+      clusterName = clusterName.slice(0, portIndex);
+    }
+  }
+  return clusterName;
+};
+
+const getOpenshiftVersion = openshiftVersionResponse => {
+  const result = get(openshiftVersionResponse, 'data.result');
+  let version;
+  if (result) {
+    let item = result;
+    if (Array.isArray(result)) {
+      [item] = result;
```

Comment:
can we use more obvious syntax here, please?
Sort of
```
item = result[0]
```

Reply:
```
const item = Array.isArray(result) ? result[0] : result
```

Reply:
@mareklibra updated